### PR TITLE
Add support for XOAUTH2

### PR DESCRIPTION
**Pull request by ChristophWurst**
This is the foundation for https://github.com/nextcloud/mail/issues/6454 and https://github.com/nextcloud/mail/issues/6591.

You can now add an account via CLI if you know the OAUTH access token. Horde will successfully authenticate using the token.

The process of acquiring the token will be specific to each service. We will have to implement those parts for Google and Microsoft.

### CODE REVIEW

**Review by miaulalala — COMMENTED** [review]
Do yu not need to save the auth method in the DB?

**Comment by ChristophWurst** [comment]
:goat::goat::goat::goat:  to add the migration file

File: lib/Command/CreateAccount.php [hunk 1]
```diff
@@ -115,6 +119,7 @@ protected function execute(InputInterface $input, OutputInterface $output): int
 		$account->setUserId($userId);
 		$account->setName($name);
 		$account->setEmail($email);
+		$account->setAuthMethod($authMethod);
```
[the remark this answers is not in the document]
Comment (miaulalala) [reply lib/Command/CreateAccount.php:122]: ah, didn't see the migration

**Review by miaulalala — APPROVED** [review]
LGTM! This is just for occ at the moment, right?

File: lib/Command/CreateAccount.php [hunk 2]
```diff
@@ -87,12 +88,15 @@ protected function configure() {
 		$this->addArgument(self::ARGUMENT_SMTP_SSL_MODE, InputArgument::REQUIRED);
 		$this->addArgument(self::ARGUMENT_SMTP_USER, InputArgument::REQUIRED);
 		$this->addArgument(self::ARGUMENT_SMTP_PASSWORD, InputArgument::REQUIRED);
+
+		$this->addArgument(self::ARGUMENT_AUTH_METHOD, InputArgument::OPTIONAL, '', 'password');
```
Comment (miaulalala) [inline lib/Command/CreateAccount.php:92]: Can we add a hint here what the options are?

**Comment by ChristophWurst** [comment]
> LGTM! This is just for occ at the moment, right?

Yes! https://github.com/nextcloud/mail/pull/6819#issue-1291396774

**Pull request merged**